src/sdk/namespace_merge.js

```diff
@@ -89,6 +90,13 @@ class NamespaceMerge {
     async upload_object(params, object_sdk) {
         const operation = 'ObjectCreated';
         const load_for_trigger = object_sdk.should_run_triggers({ active_triggers: this.active_triggers, operation });
+
+        if (params.chunked_content) {
+            const decoder = new ChunkedContentDecoder();
+            params.source_stream.pipe(decoder);
+            params.source_stream = decoder;
+        }
```

Comment:
Why is this needed in namespace_merge?
I would expect to do that in namespace_nb, and that other namespaces will be able to propagate the same encoding?

Reply:
when talking about S3 namespace - I couldn't find a way of using the node js aws sdk to propagate the encoded content, also using propagation solution would enforce me to find specific solution for namespace blob too (and other namespaces that are not S3). I think that this solution is the least complex.

Reply:
Keep in mind that we don't want to enforce using namespace_merge for namespace_cache cases.
So we want that all of our namespaces get the source_stream decoded (at least until we find a way to optimize) but I think I would put it in the S3 ops layer instead (meaning in s3_put_object.js or s3_utils if need to share it to other ops).

Reply:
right,
I moved the decoding to S3 ops layer (for multipart upload too).